test/ffmpeg-vaapi/transcode/avc.py

```diff
@@ -0,0 +1,158 @@
+  
+  @slash.parametrize(*gen_transcode_parameters(spec, "avc", "swhw"))
+  @platform_tags(AVC_ENCODE_PLATFORMS)
+  def test_swhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'h264',
+      case         = case,
+      mode         = 'swhw',
+      mcodec       = 'h264_vaapi',
+    )
+    self.transcode()
+
+class to_hevc(TranscoderTest):
+  @slash.parametrize(*gen_transcode_parameters(spec, "hevc", "hwhw"))
+  @platform_tags(set(AVC_DECODE_PLATFORMS) & set(HEVC_ENCODE_8BIT_PLATFORMS))
+  def test_hwhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'hevc',
+      case         = case,
+      mode         = 'hwhw',
+      mcodec       = 'hevc_vaapi',
+    )
+    self.transcode()
+
+  @slash.parametrize(*gen_transcode_parameters(spec, "hevc", "hwsw"))
+  @platform_tags(AVC_DECODE_PLATFORMS)
+  def test_hwsw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'hevc',
+      case         = case,
+      mode         = 'hwsw',
+      mcodec       = 'hevc',
+    )
+    self.transcode()
+
+  @slash.parametrize(*gen_transcode_parameters(spec, "hevc", "swhw"))
+  @platform_tags(HEVC_ENCODE_8BIT_PLATFORMS)
+  def test_swhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'hevc',
+      case         = case,
+      mode         = 'swhw',
+      mcodec       = 'hevc_vaapi',
+    )
+    self.transcode()
+
+class to_mjpeg(TranscoderTest):
+  @slash.parametrize(*gen_transcode_parameters(spec, "mjpeg", "hwhw"))
+  @platform_tags(set(AVC_DECODE_PLATFORMS) & set(JPEG_ENCODE_PLATFORMS))
+  def test_hwhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'mjpeg',
+      case         = case,
+      mode         = 'hwhw',
+      mcodec       = 'mjpeg_vaapi',
+    )
+    self.transcode()
+
+  @slash.parametrize(*gen_transcode_parameters(spec, "mjpeg", "hwsw"))
+  @platform_tags(AVC_DECODE_PLATFORMS)
+  def test_hwsw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'mjpeg',
+      case         = case,
+      mode         = 'hwsw',
+      mcodec       = 'mjpeg',
+    )
+    self.transcode()
+
+  @slash.parametrize(*gen_transcode_parameters(spec, "mjpeg", "swhw"))
+  @platform_tags(JPEG_ENCODE_PLATFORMS)
+  def test_swhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'mjpeg',
+      case         = case,
+      mode         = 'swhw',
+      mcodec       = 'mjpeg_vaapi',
+    )
+    self.transcode()
+
+class to_mpeg2(TranscoderTest):
+  @slash.parametrize(*gen_transcode_parameters(spec, "mpeg2", "hwhw"))
+  @platform_tags(set(AVC_DECODE_PLATFORMS) & set(MPEG2_ENCODE_PLATFORMS))
+  def test_hwhw(self, case):
+    vars(self).update(spec[case].copy())
+    vars(self).update(
+      dstextension = 'mpg',
```

Comment:
Please use `m2v` extension for mpeg2 output files

Reply:
changed to m2v